File: codigo.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    vector<vector<float>> v; // Lista de vértices
    vector<vector<int>> c;   // Lista de caras
    int opcion;

    while (true) {
        cout << "1. Agregar vértice ";
        cout << "2. Agregar cara ";
        cout << "3. Ver figura ";
        cout << "4. Salir ";
        cout << "Elija una opción: ";
        cin >> opcion;

        if (opcion == 1) {
            float x, y, z;
            cout << "Ingresa los valores de x, y, z: ";
            cin >> x >> y >> z;
            vector<float> vertice;
            vertice.push_back(x);
            vertice.push_back(y);
            vertice.push_back(z);
            v.push_back(vertice);
            cout << "Vértice agregado. ";
        } 
        else if (opcion == 2) {
            if (v.empty()) {
                cout << "Primero debe ingresar los vértices. ";
            } else {
                cout << "Ingresa los índices separados por espacio (termina con -1): ";
                vector<int> indices;
                int index;
                
                while (true) {
                    cin >> index;
                    if (index == -1) break; // Finaliza la entrada con -1
                    indices.push_back(index);
                }

                c.push_back(indices);
                cout << "Cara agregada. ";
            }
        } 
        else if (opcion == 3) {
            cout << "Vértices: ";
            for (int i = 0; i < v.size(); i++) {
                cout << i << " (" << v[i][0] << ", " << v[i][1] << ", " << v[i][2] << ") ";
            }

            cout << "Caras: ";
            for (int i = 0; i < c.size(); i++) {
                cout << i << " [ ";
                for (int j = 0; j < c[i].size(); j++) {
                    cout << c[i][j] << " ";
                }
                cout << "] ";
            }
        } 
        else if (opcion == 4) {
            cout << "Saliendo del sistema. ";
            break;
        } 
        else {
            cout << "Opción inválida. ";
        }
    }

    return 0;
}

File: codigo.py
v = []
c = []

while True:
    print ("1. Agregar vertice")
    print ("2. Agregar caras")
    print ("3. Ver figura")
    print ("4. Salir")
    
    opción = input("Elija una opción: ")

    if opción == "1":
        x,y,z = input("Ingresa los valores de x,y,z ").split()
        v.append((x,y,z))
        print ("Vertice agregado")
    
    elif opción == "2":
        if not v:
            print ("Primero debe ingresar los vertices")
        else:
            indices = input("Ingresa los indices separados por el espacio").split()
            c.append([int(i)for i in indices])
            print ("Cara agregada")
    
    
    elif opción == "3":
        print("Vertices")
        for i , vertice in enumerate(v):
            print(i,vertice)

        print("Caras")
        for i , cara in enumerate(c):
            print(i,cara)
        
    elif opción == "4":  
        print ("Saliendo del sistema")
        break
        
    else:
        print ("Opción invalida")
